specs/vep-554-apache-airflow-integration/README.MD

```diff
@@ -377,7 +377,14 @@ Unit and integration tests will be introduced by following the official Airflow
 
 The communication between VDK Provider and Control Service will be secured through the
 use of HTTPS and OAuth 2.0. The most commonly used OAuth 2.0 flows will be supported (
-e.g. Client Credentials Flow, Refresh Token Flow, etc.).
+e.g. Client Credentials Flow, Refresh Token Flow, etc.) by using `vdk-control-cli`'s authentication
+functionality. The functionality would be extracted as a stand-alone
+library, which would make it more concise and lightweight, avoiding the
+installation of unneeded dependencies.
+
+Having a stand-alone authentication library would also allow for the
```

Comment:
Please describe what functionality the library would provide? What is the interface of the library (in other words What is the expected input and output)
Let's finalize it here. 

I am thinking something along those lines: 

It would provide an Authentication class to which we can pass configuration - we will reuse https://github.com/vmware/versatile-data-kit/blob/main/projects/vdk-control-cli/src/vdk/internal/control/auth/auth.py#L84 class here. 

We will extend the class also to support necessary features: 
 A) Because in airflow we may not have local disk access always or it may not be considered secure we will introduce the option to keep the Autheorization Cache in memory 
B) THis is a question for you. do we need to encrypt the keys? You mentioned below it's necessary ?

Reply:
Extended the description of the expected functionality.